src/mame/audio/8080bw.cpp

```diff
@@ -1397,3 +1397,250 @@ WRITE8_MEMBER( _8080bw_state::darthvdr_08_w )
 
 	m_port_1_last_extra = data;
 }
+
+
+/*********************************************************/
+/*                                                       */
+/* Model Racing "Cane" (Slightly based on Claybuster hw) */
+/*                                                       */
+/*********************************************************/
+
+void _8080bw_state::cane_audio(machine_config &config)
+{
+  m_cane_vco_rc_chargetime = INT_MAX;
+
+  TIMER(config, "cane_vco_timer").configure_periodic(FUNC(_8080bw_state::cane_vco_voltage_timer), attotime::from_hz(1000));
+
+  SPEAKER(config, "mono").front_center();
+
+  SN76477(config, m_sn);
+
+  // Amplitude res in the schematic is connected to a 470K potentiometer, so from the schematic is impossible to know the real res.
+  // This parameter drives the amp just before the audio output pin 13.
+  m_sn->set_amp_res(100+RES_K(20));
+  m_sn->set_noise_params(RES_K(39), RES_K(1), CAP_P(1000));
+  m_sn->set_decay_res(RES_M(1));
+  m_sn->set_attack_params(CAP_U(1.0), RES_K(47));  
+  m_sn->set_feedback_res(RES_K(4.7));
+  m_sn->set_vco_params(0, CAP_P(3300), RES_K(100));
+  m_sn->set_pitch_voltage(5.0);
+  m_sn->set_slf_params(CAP_U(1.0), RES_K(33));
+  m_sn->set_oneshot_params(CAP_U(10), RES_K(100));
+  m_sn->set_vco_mode(0);
+  m_sn->set_mixer_params(0, 0, 0);
+  m_sn->set_envelope_params(1, 0);
+  m_sn->set_enable(0);
+  m_sn->add_route(0, "discrete", 1.0, 0);
+
+  DISCRETE(config, m_discrete, cane_discrete);
+  m_discrete->add_route(ALL_OUTPUTS, "mono", 1.0);
+}
```

Comment:
Please use one tab per level of scope.  This applies to other places as well, e.g. I/O ports.